video/out/vo_vaapi_wayland.c

```diff
@@ -290,23 +291,30 @@ static int reconfig(struct vo *vo, struct mp_image_params *params)
     struct priv *p = vo->priv;
     struct vo_wayland_state *wl = vo->wl;
 
-    if (!p->solid_buffer_pool) {
-        int width = 1;
-        int height = 1;
-        int stride = MP_ALIGN_UP(width * 4, 16);
-        int fd = vo_wayland_allocate_memfd(vo, stride);
-        if (fd < 0)
-            return VO_ERROR;
-        p->solid_buffer_pool = wl_shm_create_pool(wl->shm, fd, height * stride);
-        if (!p->solid_buffer_pool)
-            return VO_ERROR;
-        p->solid_buffer = wl_shm_pool_create_buffer(p->solid_buffer_pool, 0,
-                                                    width, height, stride,
-                                                    WL_SHM_FORMAT_XRGB8888);
-        if (!p->solid_buffer)
-            return VO_ERROR;
+    if (!p->solid_buffer) {
+        /* Use single-pixel if it exists, falling back to wl_shm */
+        if (wl->single_pixel_manager) {
+            p->solid_buffer = wp_single_pixel_buffer_manager_v1_create_u32_rgba_buffer(
+                wl->single_pixel_manager, 0, 0, 0, UINT32_MAX); /* R, G, B, A */
+        } else  {
```

Comment:
We probably only want to do this `if (!p->solid_buffer_pool)`?

Reply:
Oh I guess the outer `!p->solid_buffer` check is equivalent here.

Reply:
Yes originally I had `if (wl->single_pixel_manager) && (!p->solid_buffer)` but that was because I misread the original code. That conditional recreates solid buffer on each reconfigure which doesn't seem to be necessary. I think it should always be true that `!p->solid_buffer-pool` implies `!p->solid_buffer`.